README.md

```diff
@@ -89,6 +89,36 @@ export default {
 </script>
 ```
 
+## Injecting another instance of highlight.js
+
+If you are using a different version of `highlight.js` package than this package, you may end up with two different instances of the `hljs` singleton. You can explicitly inject your instance of `hljs` into this plugin as follows:
+
+```vue
+<script lang="ts">
+import { defineComponent, provide } from 'vue'
+import hljsPlugin from '@highlightjs/vue-plugin'
+import hljs from 'highlight.js'
+
+export default defineComponent({
+    setup() {
+        provide(hljsPlugin.HLJS_INSTANCE_KEY, hljs)
```

Comment:
My first thought was doing the injection via the plugins existing `.use()` call with `options` that are passed through to `.install()`.  Can you see any advantages/disadvantages between this and provide/inject?

Reply:
i.e. `app.use(vuePlugin, { hljs })`

Reply:
It's to provide compatibility with users that choose to import this component directly rather than installing globally

The downside is that the corresponding `provide` call must be called inside a Vue component (either the immediate parent or the global root App)

Reply:
> choose to import this component directly rather than installing this plugin globally

Supporting both seems to make sense - and seems like a reasonable thing - does the Vue ecosystem have no convention for doing this?  This is exactly the kind of discussion I was hoping might happen (discussion of caveats that I wouldn't know about)... sorry if this sounds dumb:

> What if we just did it the plain old JS way?  What if the plugin exports a function that you call and it returns the component... 

```
var component = hljsVuePlugin()
var component = hljsVuePlugin({instance: externalHLJS})
```

That's what I would have done initially if I'd have tackled this, lol...

Reply:
Other libraries may use factory functions internally but I haven't seen any that actually export the functions. The Vue plugins that I've seen generally rely on `app.use` to provide custom options globally. If they are imported/used locally, they rely on props.

The more I think about this niche use case, the more I think this PR might be unnecessary. Ultimately, this issue stems from incompatible `highlight.js` semver resulting in 2 different file locations `node_modules/highlight.js` and `node_modules/@highlight.js/vue-plugin/node_modules/highlight.js`. This can be resolved in a number of ways outside of the Vue ecosystem:

- Create an issue asking us to update this project's `package.json` (Due to bad timing, I wrote this Vue 3 version right before highlight.js 11 came out which caused this chain of events)
- Update `highlight.js`'s dependency with `>= last compatible version < 12` (for the odd use case where someone can't update to 11 yet?)
- Use yarn's resolutions to overwrite which version of `highlight.js` to use
- Use webpack's `resolve` option to overwrite where to import `highlight.js` when bundling the code

I think it might be enough to just add a note in the README about the problem with different versions. If they're incompatible semvers, there's a good chance there's a breaking change so all these solutions may not even work vs simply opening an issue to update the package.json.

Reply:
Yeah, the people getting caught out by this are still going to get caught out by this after adding optional injection. You don't care about injection until the issue occurs. Then you just need a way to fix it, which you can do by sorting out the version mismatch rather than injection (unless there happens to be some forced versioning in vue-plugin you actually wanted to override, but that's a niche niche case).

On the other hand, enforced injection or peer dependency of hljs could remove the head scratcher issue. In it's place would be a hard error if the dev messes up. So vue-plugin doesn't include the dependency and you must install `highlight.js` in the parent project. `yarn add highlight.js @hljs/vue-plugin`

The peer dependency would look like it does currently:
```
import { hljsComponent, hljsVuePlugin } from '@hljs/vue-plugin' // this would fail if dev hasn't installed hljs
```
or enforced injection (but this looks a bit pointless writing it out after above):
```
import hljs from 'highlight.js'
import { hljsVueSetup } from '@hljs/vue-plugin'
const { hljsComponent, hljsVuePlugin } = hljsVueSetup(hljs) // this would fail if hljs wasn't provided
```

Reply:
I forgot about peerDependencies. It seems to be made for this purpose

Reply:
If we all agree `peerDependencies` could work then pinning that to `>= 11.x` (or so) I think would be the simplest fix.  It doesn't 100% feel like a peer dependency (vs direct) to me but I can certainly see how declaring it as such could be beneficial.